server/auth/sso/sso.go

```diff
@@ -280,6 +293,18 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+	if s.filterGroupsRegex != nil && len(s.filterGroupsRegex) > 0 {
```

Comment:
```suggestion
        // only return groups that match at least one of the regexes
	if s.filterGroupsRegex != nil && len(s.filterGroupsRegex) > 0 {
```
a comment would be useful to describe the logic here. that would be consistent with the above code and help readers as the code here is very nested

Reply:
done